src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php

```diff
@@ -0,0 +1,171 @@
+        }
+        $io->newLine();
+
+        /** @var ChildDefinition $autoconfiguredInstanceofItem */
+        foreach ($autoconfiguredInstanceofItems as $key => $autoconfiguredInstanceofItem) {
+            $tableRows = [];
+
+            foreach ($autoconfiguredInstanceofItem->getTags() as $tag => $tagAttributes) {
+                $tableRows[] = ['Tag', $tag];
+                if ($tagAttributes !== [[]]) {
+                    $tableRows[] = ['Tag attribute', $this->dumpTagAttribute($tagAttributes)];
+                }
+            }
+
+            if ($autoconfiguredInstanceofItem->getMethodCalls()) {
+                $tableRows[] = ['Method call', $this->dumpMethodCall($autoconfiguredInstanceofItem)];
+            }
+
+            if ($autoconfiguredInstanceofItem->getBindings()) {
+                $tableRows[] = ['Bindings', $this->dumpBindings($autoconfiguredInstanceofItem)];
+            }
+
+            $io->writeln(sprintf("Autoconfiguration for \"%s\"\n==============================================", $key));
+            $io->newLine();
+            $io->table(['Option', 'Value'], $tableRows);
+        }
+    }
+
+    private function dumpMethodCall(ChildDefinition $autoconfiguredInstanceofItem)
+    {
+        $tagContainerBuilder = new ContainerBuilder();
+        foreach ($tagContainerBuilder->getServiceIds() as $serviceId) {
+            $tagContainerBuilder->removeDefinition($serviceId);
+            $tagContainerBuilder->removeAlias($serviceId);
+        }
+        $tagContainerBuilder->addDefinitions([$autoconfiguredInstanceofItem]);
+
+        $dumper = new YamlDumper($tagContainerBuilder);
+        preg_match('/calls\:\n((?: +- .+\n)+)/', $dumper->dump(), $matches);
+
+        return preg_replace('/^\s+/m', '', $matches[1]);
+    }
+
+    private function dumpBindings(ChildDefinition $autoconfiguredInstanceofItem)
+    {
+        $tagContainerBuilder = new ContainerBuilder();
+        foreach ($tagContainerBuilder->getServiceIds() as $serviceId) {
+            $tagContainerBuilder->removeDefinition($serviceId);
+            $tagContainerBuilder->removeAlias($serviceId);
+        }
+
+        $dumper = new YamlDumper($tagContainerBuilder);
+        foreach ($autoconfiguredInstanceofItem->getBindings() as $bindingKey => $bindingValue) {
+            $tagContainerBuilder->setParameter($bindingKey, $bindingValue->getValues()[0]);
+        }
+
+        preg_match('/parameters\:\n((?: + .+\n)+)/', $dumper->dump(), $matches);
+
+        return preg_replace('/^\s+/m', '', $matches[1]);
+    }
+
+    private function dumpTagAttribute(array $tagAttribute)
+    {
+        $cloner = new VarCloner();
+        $cliDumper = new CliDumper(null, null, AbstractDumper::DUMP_LIGHT_ARRAY);
+
+        return $cliDumper->dump($cloner->cloneVar(current($tagAttribute)), true);
+    }
+
+    /**
+     * @return ContainerBuilder
+     */
+    private function getContainerBuilder()
```

Comment:
is the version from `ContainerDebugCommand::getContainerBuilder()` blocking for this implementation? extending from `getContainerBuilder` gives us things for free